CMakeLists.txt

```diff
@@ -72,16 +72,23 @@ endif()
 
 # standard dependency macro
 macro(Cabana_add_dependency)
-  cmake_parse_arguments(CABANA_DEPENDENCY "" "PACKAGE;VERSION" "" ${ARGN})
-  find_package( ${CABANA_DEPENDENCY_PACKAGE} ${CABANA_DEPENDENCY_VERSION} QUIET )
+  cmake_parse_arguments(CABANA_DEPENDENCY "" "PACKAGE;VERSION;COMPONENTS" "" ${ARGN})
+  # Workaround for using HDF5 C-bindings
+  if(${CABANA_DEPENDENCY_PACKAGE} STREQUAL HDF5)
```

Comment:
```suggestion
  if(CMAKE_VERSION VERSION_LESS 2.26 AND CABANA_DEPENDENCY_PACKAGE STREQUAL "HDF5")
```

Reply:
I talked to Kitware, and one can not disable a language, but he did merge a change to findHDF5 where it does not depend on C being enabled, https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8015/diffs

It should be in the next CMake release, so at that time, you can check the CMake version and not load the C language.